### PR TITLE
Add contribution guidelines

### PR DESCRIPTION
**Pull request by InfinityGhost**
Added contribution guidelines, along with associated branch rules.

Note: All newly created pull requests must abide to these new guidelines once this is merged.

### CODE REVIEW

**Comment by EXtremeExploit** [comment]
what about end of line at the end of files?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+# Code Contributions
+
+We strive to keep a maintainable and easily readable codebase in order to keep OpenTabletDriver alive.
+
+The following rule(s) are applicable to all code contributions:
+
+- The git branch should be based on and targeting the `master` branch (except under certain conditions)
+- All trailing whitespace must be trimmed before committing.
+- Follow the language's documented formatting guidelines.
```
Comment (Sublimelime) [inline CONTRIBUTING.md:59]: I would add a link to said guidelines (are we using Microsoft's?) or an explanation of what we do, like the whole "curly brackets on their own line" and "space between 'if' and condition", etc

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+- Use spaces for indentation, do not use the tab character.
+
```
Comment (Sublimelime) [inline CONTRIBUTING.md:61]: I would add:

- Avoid merging master into your branch, unless it's to fix a merge conflict
- If your PR fixes an issue, include the keyword to make it auto-close the issue on merge ("fixes \#number")
- Keep PRs to one fix/purpose, to ease merging

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+When adding a new tablet configuration file to the `OpenTabletDriver.Configurations` project, these rules must be followed.
+
+- The `TABLETS.md` file must be updated to include the newly added configuration, including any quirks or missing features.
+- The tablet name should follow `[Manufacturer] [Model Number/Product Name]`
+- The file name must be the tablet name with the `.json` extension.
+- The file must be located in the directory designated to an individual manufacturer.
+- All trailing whitespace must be trimmed before committing.
+  > When using the configuration editor within the OpenTabletDriver UI, this will be performed automatically.
+- The current formatting of the `TABLETS.md` document must be strictly followed as below:
+
```
Comment (Sublimelime) [inline CONTRIBUTING.md:27]: I would add 

> Include contact info of the person asking for support, such as an @ for their account on github, a discord username, etc, for functionality verification"

**Review by Sublimelime — CHANGES_REQUESTED** [review]

**Review by Sublimelime — APPROVED** [review]

**Review by MishterKirby — APPROVED** [review]

**Review by jamesbt365 — APPROVED** [review]

**Pull request merged**